File: CoreAutomata/GameAreaManager.cs
namespace CoreAutomata
{
    public static class GameAreaManager
    {
        public static void Reset()
        {
            ScriptDimension = CompareDimension = null;
            _gameArea = null;
        }

        static Region _gameArea;

        public static Region GameArea
        {
            get => _gameArea ??= AutomataApi.WindowRegion;
            set => _gameArea = value;
        }

        /// <summary>
        /// Dimensions at which script was written. Compare dimension can be smaller for faster image comparison.
        /// </summary>
        public static CompareSettings ScriptDimension { get; set; }

        /// <summary>
        /// Image Matching dimension. All images, regions, locations are scaled to this to keep the script device independent.
        /// </summary>
        public static CompareSettings CompareDimension { get; set; }
    }
}

File: app/src/main/java/com/mathewsachin/fategrandautomata/core/GameAreaManager.kt
package com.mathewsachin.fategrandautomata.core

/**
 * This class is used for storing the script and image dimensions and for storing the playable area
 * in [GameArea].
 */
object GameAreaManager {
    /**
     * Stores either the width or height of the script dimensions. Normally, this is 1440p.
     */
    var ScriptDimension: CompareSettings? = null

    /**
     * Stores either the width or height of the image dimensions. Normally, this is 720p.
     */
    var CompareDimension: CompareSettings? = null

    private var gameArea: Region? = null

    /**
     * Stores the playable area as a [Region] in image coordinates, which is normally 720p.
     * Notches and the blue bars are excluded from this area by introducing X and Y offsets.
     */
    var GameArea: Region
        get() {
            // store the WindowRegion so we don't calculate it twice
            val windowRegion = AutomataApi.PlatformImpl.windowRegion
            return gameArea?.plus(windowRegion.location) ?: windowRegion
        }
        set(value) {
            gameArea = value
        }

    /**
     * Sets all fields to null.
     */
    fun reset() {
        ScriptDimension = null
        CompareDimension = null
        gameArea = null
    }
}